aten/src/ATen/native/TensorShape.cpp

```diff
@@ -1469,12 +1469,12 @@ Tensor trace(const Tensor& self) {
   Tensor result = at::empty({}, self.options());
   AT_DISPATCH_ALL_TYPES(self.scalar_type(), "trace", [&] {
     scalar_t sum = 0;
-    auto t_data = self.data_ptr<scalar_t>();
+    const auto* t_data = self.data_ptr<scalar_t>();
 
     int64_t i = 0;
     int64_t t_stride_0, t_stride_1, t_diag_size;
 
-    TORCH_CHECK(self.dim() == 2, "expected a matrix");
+    TORCH_CHECK(self.dim() == 2, "trace: expected a matrix, but got tensor with dim %d", self.dim());
```

Comment:
The %d doesn't actually do anything -- TORCH_CHECK concatenates everything found after the first arg. The right way to invoke it is:
```
TORCH_CHECK(self.dim() == 2, "trace: expected a matrix, but got tensor with dim ", self.dim())
```

Reply:
sorry, it's my fault